src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -4615,21 +4615,21 @@ checkNullable:
                 value = ParseExpressionCore()
 
             ElseIf modifiers.Any AndAlso modifiers.Any(SyntaxKind.OptionalKeyword) Then
-
-                equals = ReportSyntaxError(InternalSyntaxFactory.MissingPunctuation(SyntaxKind.EqualsToken), ERRID.ERR_ObsoleteOptionalWithoutValue)
-                value = ParseExpressionCore()
-
+                If CheckFeatures(Feature.ImplicitDefaultValueOnOptionalParameter, _scanner.Options) = False Then
```

Comment:
It looks to me that the method in scanner has correct implementation for the current situation (the feature is experimental) and for the future (when it becomes an official feature tied to a particular language version). So, in the long run, it would be better to use that existing helper rather than adding a new function that isn't robust to the future changes.